Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/InMemorySpawnableAssetContainer.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/string/string_view.h>
+#include <AzToolsFramework/Prefab/Spawnable/PrefabConversionPipeline.h>
+
+namespace AzToolsFramework::Prefab
+{
+    class PrefabSystemComponentInterface;
+    class PrefabLoaderInterface;
+}
+
+namespace AzToolsFramework::Prefab::PrefabConversionUtils
+{
+
+    class InMemorySpawnableAssetContainer
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(InMemorySpawnableAssetContainer, AZ::SystemAllocator, 0);
+
+        using Assets = AZStd::vector<AZ::Data::Asset<AZ::Data::AssetData>>;
+        using CreateSpawnableResult = AZ::Outcome<AZ::Data::Asset<AZ::Data::AssetData>&, AZStd::string>;
+        using RemoveSpawnableResult = AZ::Outcome<void, AZStd::string>;
+
+        struct SpawnableAssetData
+        {
+            Assets m_assets;
+            AZ::Data::AssetId m_spawnableAssetId;
+        };
+        using SpawnableAssets = AZStd::unordered_map<AZStd::string, SpawnableAssetData>;
+
+        virtual ~InMemorySpawnableAssetContainer();
```

Comment:
There are no other virtual functions and it doesn't seem like this class is designed to be inherited from so the `virtual` doesn't seem needed.